src/main/scala/is/hail/methods/MinHash.scala

```diff
@@ -0,0 +1,64 @@
+package is.hail.methods
+
+import breeze.linalg.{DenseMatrix, DenseVector, min}
+import is.hail.variant.{Genotype, Variant, VariantDataset}
+import is.hail.annotations.Annotation
+import scala.collection.mutable.HashSet
+import scala.util.hashing.{MurmurHash3 => MH3}
+import scala.util.Random
+
+object MinHash {
+  // returns a k by vds.nSamples matrix
+  def kMinHash(vds: VariantDataset, k: Int): DenseMatrix[Int] = {
+    val n = vds.nSamples
+    def seqOp(l: DenseMatrix[Int], r: (Variant, (Annotation, Iterable[Genotype]))): DenseMatrix[Int] = r match {
+      case (_, (_, gs)) =>
+        val gsItr = gs.hardCallGenotypeIterator
+        val hashes = DenseVector.fill[Int](k)(Random.nextInt())
+        var i = 0
+        while(i < n) {
+          //TODO: Write a min UFunc that modifies the left arg in place
+          if (gsItr.next() > 0) l(::,i) := min(l(::,i), hashes)
+          i += 1
+        }
+        l
+    }
+    val zeroValue = DenseMatrix.fill[Int](k, n)(Int.MaxValue)
+    vds.rdd.aggregate(zeroValue)(seqOp, min(_,_))
```

Comment:
I take it back; `min.apply` doesn't work either. (It satisfied IntelliJ, but not scalac).